CMakeLists.txt

```diff
@@ -1719,7 +1719,31 @@ endif()
 
 # Generate and use MCUboot related artifacts as needed.
 if(CONFIG_BOOTLOADER_MCUBOOT)
-  include(${CMAKE_CURRENT_LIST_DIR}/cmake/mcuboot.cmake)
+  get_property(signing_script_set
```

Comment:
We have a zephyr property target for handling Zephyr build system properties, partly because it allows extracting such properties using generator expressions.

https://github.com/zephyrproject-rtos/zephyr/blob/7f0774397ed2924d21e7679ad8f452689ed7eba7/cmake/modules/kernel.cmake#L200-L205

In this case it would also remove the need for `get_property(... SET)` followed by another `get_property(...)`, cause fetching a target property returns `<var>-NOTFOUND` when not set, so test simply becomes:
```
get_target_property(signing_script zepyr_property SIGNING_SCRIPT)
if(NOT signing_script)
  set(signing_script ${CMAKE_CURRENT_LIST_DIR}/cmake/mcuboot.cmake)
  set_target_properties(zepyr_property PROPERTIES SIGNING_SCRIPT ${signing_script})
endif()

message(STATUS "Using signing script: ${signing_script}")

include(${signing_script})
```

Reply:
Neat, have updated to use that method instead.